### PR TITLE
python311Packages.fslpy: init at 3.21.0

### PR DESCRIPTION
**Pull request by bcdarwin**
## Description of changes

Init python311Packages.fslpy, a [Python programming library for interacting with the FSL neuroimaging software suite](https://git.fmrib.ox.ac.uk/fsl/fslpy). Note that the functionality of this package is limited since FSL and its atlases are not available in Nixpkgs (unfree license and difficult to install) but the user can always download the atlases, and other functionality is still useful e.g. interacting with image and transform files, and should be able to be used as a dependency for the standalone open source medical image viewer FSLeyes.

Doesn't build on Python 3.12 due to the optional wxPython dependency which is used for tests. I could have patched this out but it doesn't seem worthwhile since FSLeyes itself also depends on wxPython.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Sigmanificient — APPROVED** [review]

**Review by Sigmanificient — APPROVED** [review]

File: pkgs/development/python-modules/fslpy/default.nix [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      indexed-gzip
+      pillow
+      rtree
+      trimesh
+      wxpython
+    ];
+  };
+
+  nativeCheckInputs = [
+    coverage
```
[the remark this answers is not in the document]
Comment (bcdarwin) [reply pkgs/development/python-modules/fslpy/default.nix]: Removed.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 329952` at eed29b0b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>python311Packages.fslpy</li>
  </ul>
</details>

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 329952` at eed29b0b run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>python311Packages.fslpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/fslpy/default.nix [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "fslpy";
+  version = "3.21.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchgit {
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/fslpy/default.nix:29]: Use fetchFromGitLab with domain = "https://git.fmrib.ox.ac.uk"

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/fslpy/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/fslpy/default.nix]: Switched.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**